# do not assume '.' being in PATH on hpux,
# use 'gcc' as linker, link with '-lm'.

--- hints/hpux.sh
+++ hints/hpux.sh
@@ -190,7 +190,7 @@
     $define|true|[Yy])
 	echo '#include <stdio.h>\nint main(){long l;printf("%d\\n",sizeof(l));}'>try.c
 	$cc -o try $ccflags $ldflags try.c
-	if [ "`try`" = "8" ]; then
+	if [ "`./try`" = "8" ]; then
 	    case "$use64bitall" in
 		$define|true|[Yy]) ;;
 		*)  cat <<EOM >&4
@@ -338,7 +338,7 @@
     } /* main */
 EOF
 $cc -o try $ccflags $ldflags try.c
-	maxdsiz=`try`
+	maxdsiz=`./try`
 rm -f try try.c core
 if [ $maxdsiz -le 64 ]; then
     # 64 Mb is probably not enough to optimize toke.c
@@ -363,11 +363,12 @@
 	    "")           optimize="-g -O" ;;
 	    *O[3456789]*) optimize=`echo "$optimize" | sed -e 's/O[3-9]/O2/'` ;;
 	    esac
-	#ld="$cc"
-	ld=/usr/bin/ld
+	ld="$cc"
+	#ld=/usr/bin/ld
 	cccdlflags='-fPIC'
-	#lddlflags='-shared'
-	lddlflags='-b'
+	lddlflags='-shared'
+	#lddlflags='-b'
+	libs='-lm'
 	case "$optimize" in
 	    *-g*-O*|*-O*-g*)
 		# gcc without gas will not accept -g
